From fe063b162411651213b87e2f7eda4ce8c1db0e27 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Fri, 13 Oct 2017 15:48:31 +0100
Subject: [PATCH 6/9] intercept: Blacklist games from replacing
 security-sensitive libcurl

As seen in DiRT Rally, the Feral Interactive port is replacing libcurl.so.4,
which is a security sensitive library, known to update against CVEs with
frightening frequency.

Protect our users, and ensure system libcurl is used.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 src/intercept/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/intercept/main.c b/src/intercept/main.c
index b8a01ae..fe4f49a 100644
--- a/src/intercept/main.c
+++ b/src/intercept/main.c
@@ -117,6 +117,9 @@ static const char *vendor_blacklist[] = {
 
         /* libglu has stable soname */
         "libGLU.so.",
+
+        /* Nuh uh. Sorry, Feral games. libcurl MUST be up to date for security! */
+        "libcurl.so.",
 };
 
 /**
-- 
2.14.2

